core/src/main/java/nl/nn/adapterframework/jms/PullingJmsListener.java

```diff
@@ -172,45 +169,6 @@ public void closeThread(Map<String,Object> threadContext) throws ListenerExcepti
 	}
 
 
-	@Override
-	public void afterMessageProcessed(PipeLineResult plr, Object rawMessageOrWrapper, Map<String, Object> threadContext) throws ListenerException {
-		super.afterMessageProcessed(plr, rawMessageOrWrapper, threadContext);
-		try {
-
-			// TODO Do we still need this? Should we rollback too? See
-			// PushingJmsListener.afterMessageProcessed() too (which does a
-			// rollback, but no commit).
-			if (!isTransacted()) {
-				if (isJmsTransacted()) {
-					// the following if transacted using transacted sessions, instead of XA-enabled sessions.
-					Session session = (Session)threadContext.get(IListenerConnector.THREAD_CONTEXT_SESSION_KEY);
-					if (session == null) {
-						log.warn("Listener ["+getName()+"] message ["+ (String)threadContext.get(PipeLineSession.messageIdKey) +"] has no session to commit or rollback");
-					} else {
-						if (plr.getState()==ExitState.SUCCESS) {
-							session.commit();
-						} else {
-							log.warn("Listener ["+getName()+"] message ["+ (String)threadContext.get(PipeLineSession.messageIdKey) +"] not committed nor rolled back either");
-							//TODO: enable rollback, or remove support for JmsTransacted altogether (XA-transactions should do it all)
-							// session.rollback();
-						}
-						if (isSessionsArePooled()) {
-							threadContext.remove(IListenerConnector.THREAD_CONTEXT_SESSION_KEY);
-							releaseSession(session);
-						}
```

Comment:
deze is nu weg?

Reply:
@nielsm5 Volgens mij is deze logic nu grotendeels in de class `BaseJmsListener`.